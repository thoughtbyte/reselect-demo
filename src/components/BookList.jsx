import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import List from "./List";

class BookList extends Component {
  render() {
    console.log("RENDERING BookList");

    return (
      <Fragment>
        <label className="label">Book List</label>
        <List list={this.props.list} />
      </Fragment>
    );
  }
}

BookList.propTypes = {
  list: PropTypes.array.isRequired,
};

function mapStateToProps(state) {
  return {
    list: state.bookList.map(item => item.bookTitle),
  };
}

export default connect(mapStateToProps)(BookList);
