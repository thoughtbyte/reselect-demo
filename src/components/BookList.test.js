import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import BookList from "./BookList";

const middlewares = []; // in case you need to use one you can just import it and add it here
const mockStore = configureStore(middlewares);
const initialState = {
  bookList: [
    { bookTitle: "Clean Code", authorName: "Robert C. Martin" },
    { bookTitle: "The Pragmatic Programmer", authorName: "Andrew Hunt" },
    { bookTitle: "You Don't Know JS", authorName: "Kyle Simpson" },
    { bookTitle: "Eloquent JavaScript", authorName: "Marijn Haverbeke" },
  ],
};
const store = mockStore(initialState);

describe("BookList", () => {
  test("should match snapshot", () => {
    const bookList = renderer
      .create(
        <Provider store={store}>
          <BookList />
        </Provider>
      )
      .toJSON();

    expect(bookList).toMatchSnapshot();
  });
});
