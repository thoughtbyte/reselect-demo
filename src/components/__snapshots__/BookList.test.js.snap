// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList should match snapshot 1`] = `
Array [
  <label
    className="label"
>
    Book List
</label>,
  <div
    className="card"
>
    <ul>
        <li>
            Clean Code
        </li>
        <li>
            The Pragmatic Programmer
        </li>
        <li>
            You Don't Know JS
        </li>
        <li>
            Eloquent JavaScript
        </li>
    </ul>
</div>,
]
`;
